contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -774,7 +776,12 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
                 IBosonFundsHandler(protocolDiamond).depositFunds{ value: balance }(sellerId, token, balance);
             } else {
                 uint256 balance = IERC20(token).balanceOf(address(this));
-                IERC20(token).approve(protocolDiamond, balance);
+                // get current allowance
```

Comment:
Similar to above, we can check if the balance  > 0 and do the other operations only then.